File: Chapter02/Exercise2_01/Exercise2_01.py
from pyspark.rdd import RDD
from pyspark.sql import SparkSession

from utilities01_py.helper_python import create_session
from utilities02_py.helper_python import sample_wet_loc

if __name__ == "__main__":
    session: SparkSession = create_session(2, "Default parsing")
    session.sparkContext.setLogLevel('ERROR')  # skips INFO messages
    records: RDD = session.sparkContext.textFile(sample_wet_loc)
    for record in records.take(50):
        print(record)
        print('-' * 20)
    print('#' * 40)
    print('Total # of records: ' + str(records.count()))

File: Chapter02/Exercise2_01/Exercise2_01.scala
package Exercise2_01

import org.apache.spark.rdd.RDD
import org.apache.spark.sql.SparkSession
import Utilities01.HelperScala.createSession
import Utilities02.HelperScala.sampleWetLoc

object Exercise2_01 {

  def main(args: Array[String]) {
    val session: SparkSession = createSession(2, "Default crawl parsing")
    session.sparkContext.setLogLevel("ERROR") // skips INFO messages
    val records: RDD[String] = session.sparkContext.textFile(sampleWetLoc)
    // code for subsequent steps comes here


    records.take(50).foreach(record => {
      println(record)
      println("-" * 20)
    })
    println("#" * 40)
    println(s"Total # of records: ${records.count()}")
  }
}
